src/SourceBuild/tarball/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/DotNetFormatTests.cs

```diff
@@ -0,0 +1,104 @@
+using System;
+using System.Linq;
+
+namespace Microsoft.DotNet.SourceBuild.SmokeTests;
+
+public class DotNetFormatTests
+{
+    const string VerifyNoChangesTestDirectoryName = "VerifyNoChanges";
+    const string FormatTestDirectoryName = "FormatTest";
+    const string UnformattedDirectoryName = "unformatted";
+    const string SolutionDirectoryName = "solution";
+    const string TestFileName = "Test.cs";
+
+    private ITestOutputHelper _outputHelper { get; }
+    private DotNetHelper _dotNetHelper { get; }
+    private string _formatTestRootDirectory { get; }
+
+    public DotNetFormatTests(ITestOutputHelper outputHelper)
+    {
+        this._outputHelper = outputHelper;
+        this._dotNetHelper = new DotNetHelper(outputHelper);
+        this._formatTestRootDirectory = Path.Combine(Directory.GetCurrentDirectory(), "format");
+    }
+
+    /// <Summary>
+    /// Create a new project and verify that it doesn't need to be formatted.
+    /// </Summary>
+    [Fact]
+    public void DotNetFormatVerifyNoChanges()
+    {
+        string verifyNoChangesTestPath = Path.Join(this._formatTestRootDirectory, VerifyNoChangesTestDirectoryName);
+
+        CreateNewProject(verifyNoChangesTestPath);
+
+        string verifyNoChangesCsprojPath = Path.Combine(verifyNoChangesTestPath, VerifyNoChangesTestDirectoryName + ".csproj");
+
+        string[] dotnetFormatArgs = {
+            "format",
+            verifyNoChangesCsprojPath,
+            "--verify-no-changes"
+        };
+
+        _dotNetHelper.ExecuteDotNetCmd(String.Join(' ', dotnetFormatArgs), _outputHelper);
+    }
+
+    /// <Summary>
+    /// Format an unformatted project and verify that the output matches the
+    /// pre-computed solution.
+    /// </Summary>
+    [Fact]
+    public void DotNetFormatProject()
+    {
+        string formatTestPath = Path.Join(this._formatTestRootDirectory, FormatTestDirectoryName);
+        string formatTestCsprojPath = Path.Join(formatTestPath, FormatTestDirectoryName + ".csproj");
+
+        string unformattedCsFilePath = Path.Combine(this._formatTestRootDirectory, UnformattedDirectoryName, TestFileName);
+        string solutionCsFilePath = Path.Combine(this._formatTestRootDirectory, SolutionDirectoryName, TestFileName);
+        string testCsFilePath = Path.Combine(formatTestPath, TestFileName);
+
+        CreateNewProject(formatTestPath);
+
+        File.Copy(unformattedCsFilePath, testCsFilePath);
+
+        string[] dotnetFormatArgs = {
+            "format",
+            formatTestCsprojPath,
+        };
+
+        _dotNetHelper.ExecuteDotNetCmd(String.Join(' ', dotnetFormatArgs), _outputHelper);
+
+        Assert.True(File.ReadAllLines(testCsFilePath).SequenceEqual(File.ReadAllLines(solutionCsFilePath)));
+    }
+
+    /// <Summary>
+    /// Run `dotnet new console` in projectDirectory. If projectDirectory
+    /// already exists, then empty it out first.
+    /// </Summary>
+    private void CreateNewProject(string projectDirectory)
```

Comment:
This feels like a helper method that would be useful in DotNetHelper.  It would need to be a little more abstract but I see this as being something that will be frequently called overtime.